primitives/consensus/common/src/import_queue.rs

```diff
@@ -15,292 +15,5 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Import Queue primitive: something which can verify and import blocks.
-//!
-//! This serves as an intermediate and abstracted step between synchronization
-//! and import. Each mode of consensus will have its own requirements for block
-//! verification. Some algorithms can verify in parallel, while others only
-//! sequentially.
-//!
-//! The `ImportQueue` trait allows such verification strategies to be
-//! instantiated. The `BasicQueue` and `BasicVerifier` traits allow serial
-//! queues to be instantiated simply.
-
-use std::collections::HashMap;
-
-use sp_runtime::{
-	traits::{Block as BlockT, Header as _, NumberFor},
-	Justifications,
-};
-
-use crate::{
-	block_import::{
-		BlockCheckParams, BlockImport, BlockImportParams, BlockOrigin, ImportResult, ImportedAux,
-		ImportedState, JustificationImport, StateAction,
-	},
-	error::Error as ConsensusError,
-	metrics::Metrics,
-};
-pub use basic_queue::BasicQueue;
-
-/// A commonly-used Import Queue type.
-///
-/// This defines the transaction type of the `BasicQueue` to be the transaction type for a client.
-pub type DefaultImportQueue<Block, Client> =
-	BasicQueue<Block, sp_api::TransactionFor<Client, Block>>;
-
-mod basic_queue;
-pub mod buffered_link;
-
-/// Shared block import struct used by the queue.
-pub type BoxBlockImport<B, Transaction> =
-	Box<dyn BlockImport<B, Error = ConsensusError, Transaction = Transaction> + Send + Sync>;
-
-/// Shared justification import struct used by the queue.
-pub type BoxJustificationImport<B> =
-	Box<dyn JustificationImport<B, Error = ConsensusError> + Send + Sync>;
-
-/// Maps to the Origin used by the network.
-pub type Origin = libp2p::PeerId;
-
-/// Block data used by the queue.
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct IncomingBlock<B: BlockT> {
-	/// Block header hash.
-	pub hash: <B as BlockT>::Hash,
-	/// Block header if requested.
-	pub header: Option<<B as BlockT>::Header>,
-	/// Block body if requested.
-	pub body: Option<Vec<<B as BlockT>::Extrinsic>>,
-	/// Indexed block body if requested.
-	pub indexed_body: Option<Vec<Vec<u8>>>,
-	/// Justification(s) if requested.
-	pub justifications: Option<Justifications>,
-	/// The peer, we received this from
-	pub origin: Option<Origin>,
-	/// Allow importing the block skipping state verification if parent state is missing.
-	pub allow_missing_state: bool,
-	/// Skip block exection and state verification.
-	pub skip_execution: bool,
-	/// Re-validate existing block.
-	pub import_existing: bool,
-	/// Do not compute new state, but rather set it to the given set.
-	pub state: Option<ImportedState<B>>,
-}
-
 /// Type of keys in the blockchain cache that consensus module could use for its needs.
 pub type CacheKeyId = [u8; 4];
```

Comment:
If that's the only type left here, maybe move it to top-level `lib.rs`?

Reply:
Happy to move it one level up. I'm tempted to leave it in the `import_queue` namespace for backward compatibility.

Reply:
actually I can't see it referenced in polkadot so will move it up a level.